extensions/shared/external-media/sources/with-media.js

```diff
@@ -68,7 +71,10 @@ export default function withMedia() {
 				 * This can be removed once
 				 * https://github.com/WordPress/gutenberg/issues/22940 is fixed.
 				 */
-				if ( [ UP, DOWN, LEFT, RIGHT ].includes( event.keyCode ) ) {
+				if (
+					[ UP, DOWN, LEFT, RIGHT ].includes( event.keyCode ) &&
+					! this.contentElement.contains( event.target ) // Let event through to enable arrow keys navigation.
```

Comment:
I found that this does not stop propagation on the pexels search input or search button. So when you press the arrow keys on those, propagation is not stopped and the modal closes. It doesn't close when using arrow keys on the media items because those call `stopPropagation` separately within `handleMediaItemKeyDown`. 

The search input and button do not have anything that calls `stopPropogation`, so the modal closes on arrow key presses. I think we should change this check to _only let arrow key presses avoid this `stopPropogation` if the event target is a media item_. I'm not sure a non-hacky way to do this though :/ It'd be nice to use refs, but I don't think that will be possible.